### PR TITLE
squeezelite: restructure package variants

### PR DESCRIPTION
**Pull request by robho**
The squeezelite packages were lacking some features/codecs (ssl, opus, wma, dsd and resampling) and required manual compilation to enable these features/codecs.

Now there are 3 packages available that hopefully should satisfy most users:
* full - This package enables all features and codecs.
* dynamic - This package only enables PCM/AIFF and MP3 codecs and all other features/codecs can be added by manually installing OpenWrt packages.
* custom - This package allows for customizing squeezelite.

@neheb, I dropped dependency on BUILD_PATENTED which you added in 10c087a98c2ab8e5dcf569a052987b10e3e582e4. I removed the dependency because it looks like BUILD_PATENTED only controls which video codecs to include in ffmpeg and squeezelite uses libffmpeg-audio-dec which doesn't contain any video codecs. Does this seem ok or should I restore the dependency on BUILD_PATENTED?

Maintainer: @thess 
Compile tested: ath79, D-Link DIR-825, 22.03 and master
Run tested: ath79, D-Link DIR-825, 22.03, audio playback works with different codecs

### CODE REVIEW

**Comment by neheb** [comment]
Hmmm WMA seems to not be patented. Here's a relatively full list: https://src.fedoraproject.org/rpms/ffmpeg/blob/rawhide/f/enable_decoders

File: sound/squeezelite/Makefile [hunk 1]
```diff
@@ -6,13 +6,13 @@
 include $(TOPDIR)/rules.mk
 
 PKG_NAME:=squeezelite
-PKG_VERSION:=1.9.9-1415
+PKG_VERSION:=1.9.9-1432
```
Comment (BKPepe) [inline sound/squeezelite/Makefile:9]: This update should be done in separate commit.

Comment (robho) [reply sound/squeezelite/Makefile]: OK, I created #21096

**Comment by robho** [comment]
This PR now depends on #21096 which brings a fix for compiling with OpenSSL 3

**Comment by neheb** [comment]
Wonder what the difference is between mad and mpg123 here. The former should be smaller.

File: sound/squeezelite/Makefile [hunk 2]
```diff
@@ -28,104 +28,185 @@ define Package/squeezelite/default
...
-
-	config SQUEEZELITE_OPUS
-	    bool "Opus codec support"
-	    help
-		Include Opus codec support
-	    default n
+    TITLE+= (dynamic)
+    VARIANT:=dynamic
+    DEPENDS+= +libmpg123
+    PKG_BUILD_DEPENDS:=faad2 ffmpeg flac libsoxr libvorbis \
```
[the remark this answers is not in the document]
Comment (robho) [reply sound/squeezelite/Makefile:47]: The dynamic variant still contains code to decode aac, alac, flac, ogg, opus, ... by using dynamically loaded libraries. The code needed to interface with the decode libraries is small, but it uses data structures defined in the decode libraries and that's why I added PKG_BUILD_DEPENDS so that header files are available when building the decode "stubs".

[the remark this answers is not in the document]
Comment (neheb) [reply sound/squeezelite/Makefile:47]: Does PKG_BUILD_DEPENDS even work when added here?

**Comment by neheb** [comment]
So mad is faster on mips and slower everywhere else. It's smaller everywhere. No real opinion on whether or not it should be default. It's an older library that decodes to int instead of floating point.

File: sound/squeezelite/Makefile (hunk 2 above)
[the remark this answers is not in the document]
Comment (robho) [reply sound/squeezelite/Makefile]: No, it didn't work, thanks! :-)

I moved it to the top of the file. I think the only negative side effect of this is that a slimmed down custom variant build will build dependencies that aren't needed.

Is that ok or do you see any other way of handling this?

[the remark this answers is not in the document]
Comment (neheb) [reply sound/squeezelite/Makefile]: I find no issue with that.
